### PR TITLE
Changes the way xeno abilities are and buffs defender at ancient

### PR DESCRIPTION
**Pull request by Maciekkub**
## About The Pull Request
This PR basically focuses itself on the idea that now there's room for customization in terms of abilities for various maturation stages for every caste, it is done by some changes in the code i am not exactly sure how to explain that, although it is clearly visible that now the abilities are in the castedatum file and you can change them for various maturation stages. An example of this is that Defender at Ancient now gets the lunge ability in place of it's forward charge. This supposedly adheres for xenos to not crave T3 and makes surviving until Ancient more rewarding.
Also huge thanks to Hughent!

## Why It's Good For The Game
Adds possible room for customization of abilities based on the maturation stage as well as buffs defender because it's a not a strong caste itself.

## Changelog
:cl:
balance: Defender caste upon reaching Ancient maturation stage has its forward charge replaced by warrior's lunge
:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 1]
```diff
@@ -17,15 +17,15 @@
 	var/obj/item/explosive/grenade/grenade_type = "/obj/item/explosive/grenade/xeno"
 	var/datum/effect_system/smoke_spread/xeno/smoke
 
-	actions = list(
+	/*actions = list(
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm]: Remove all instances of commented out code.

**Comment by DominikPanic** [comment]
Also make the changelog more human readable please. Refactors are not useful for the users to know unless behavior is changed, which your other entry does.

**Comment by DominikPanic** [comment]
Again, users are not programmers. The changelog entry doesn't tell them anything important since it doesn't change their gameplay. Just keep the balance one there.

**Comment by Maciekkub** [comment]
> Again, users are not programmers. The changelog entry doesn't tell them anything important since it doesn't change their gameplay. Just keep the balance one there.

Should i also change the description and the Why is it good for the game to suit it being more of a rebalance thing rathen than a refactor?

**Comment by DominikPanic** [comment]
No, the description is for the devs, the changelog is for the users. It's okay to be technical there.

**Comment by Maciekkub** [comment]
> No, the description is for the devs, the changelog is for the users. It's okay to be technical there.

I see, alright. I'll keep an eye on that in the future

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm [hunk 2]
```diff
@@ -47,6 +47,22 @@
...
+	actions = list(
+		/datum/action/xeno_action/xeno_resting,
+		/datum/action/xeno_action/regurgitate,
+		/datum/action/xeno_action/plant_weeds,
+		/datum/action/xeno_action/choose_resin,
+		/datum/action/xeno_action/activable/secrete_resin,
+		/datum/action/xeno_action/grow_ovipositor,
+		/datum/action/xeno_action/activable/screech,
+		/datum/action/xeno_action/activable/corrosive_acid/strong,
+		// /datum/action/xeno_action/activable/gut, We're taking this out for now.
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm]: Commented out code

**Comment by DominikPanic** [comment]
This PR has not been tested. A lot of abilities not working, locked, or on cooldown.

```
[21:18:19] Runtime in abilities_runner.dm, line 27: Cannot read null.savage
proc name: update button icon (/datum/action/xeno_action/toggle_savage/update_button_icon)
src: Toggle Savage (/datum/action/xeno_action/toggle_savage)
call stack:
Toggle Savage (/datum/action/xeno_action/toggle_savage): update button icon()
Mature Runner (519) (/mob/living/carbon/xenomorph/runner): update action button icons()
Mature Runner (519) (/mob/living/carbon/xenomorph/runner): Life(2, 233)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm (hunk 2 above)
Comment (Maciekkub) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm]: That was there before i made any changes, should i still remove it?

**Comment by Maciekkub** [comment]
> This PR has not been tested. A lot of abilities not working, locked, or on cooldown.
> 
> ```
> [21:18:19] Runtime in abilities_runner.dm, line 27: Cannot read null.savage
> proc name: update button icon (/datum/action/xeno_action/toggle_savage/update_button_icon)
> src: Toggle Savage (/datum/action/xeno_action/toggle_savage)
> call stack:
> Toggle Savage (/datum/action/xeno_action/toggle_savage): update button icon()
> Mature Runner (519) (/mob/living/carbon/xenomorph/runner): update action button icons()
> Mature Runner (519) (/mob/living/carbon/xenomorph/runner): Life(2, 233)
> Mobs (/datum/controller/subsystem/mobs): fire(0)
> Mobs (/datum/controller/subsystem/mobs): ignite(0)
> Master (/datum/controller/master): RunQueue()
> Master (/datum/controller/master): Loop()
> Master (/datum/controller/master): StartProcessing(0)
> ```

I tested every ability for each caste and all of them have bene working properly and no runtimes have popped up: https://gyazo.com/2ea0d906d23576a392a69f8cf43380af. Could you disclose what's the problem with the abilities?

**Comment by drawsstuff** [comment]
There's been some strange issues I've been having as Xenomorph while this is test merged. Abilities get turned on and off and tables can no longer be flipped.

**Comment by Maciekkub** [comment]
> There's been some strange issues I've been having as Xenomorph while this is test merged. Abilities get turned on and off and tables can no longer be flipped.

I need to know what caste and what abilities were the ones getting turned off

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 3]
```diff
@@ -100,6 +100,8 @@
 	// *** Defiler Abilities *** //
 	var/neuro_claws_amount
 
+	var/actions = list()
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]:
```suggestion
	var/list/actions
```

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 4]
```diff
@@ -60,6 +60,7 @@
 	health = maxHealth
 	speed = xeno_caste.speed
 	armor = getArmor(arglist(xeno_caste.armor))
+	actions = xeno_caste.actions
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm]:
```suggestion
	actions = xeno_caste.actions.Copy()
```

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm]: Please do.

**Pull request merged**